source/common/common/version.cc

```diff
@@ -18,7 +18,7 @@ const std::string& VersionInfo::revisionStatus() {
 }
 
 std::string VersionInfo::version() {
-  return fmt::format("{}/{}/{}", revision(), revisionStatus(),
+  return fmt::format("{}/{}/{}/{}", revision(), ReleaseVersion, revisionStatus(),
```

Comment:
For builds in which the `git` SHA doesn't match the original SHA for the release version, we could also consider appending some suffix to indicate, e.g. `1.5.0+`.

Reply:
BTW, to figure out if we're at the same SHA as the release version, you can do something like `git rev-list -n 1  v1.5.0`, i.e. make use of the repo tags.

Reply:
This would all need to be done somewhere like https://github.com/envoyproxy/envoy/blob/2fa90d22521376a7990196e34a1e1bb945663ab4/bazel/get_workspace_status.

Reply:
It's good idea to indicate that with some suffix. I'm also willing to make use of `get_workspace_status` but it only accept specific keys as I wrote in the PR description. There might be some possible ways:

1. hard code the version number with suffix when we're developing, then drop the suffix when we release. Pros: easy and simple implementation. Cons: we need bump the version number after releases.
2. use genrule. Extract version number string from C++ header file and place somewhere else. Then extract it and pass to `git-rev-list` to figure out the SHA is same as the release version, generate a number string with suffix or without suffix and place that into C++ header file so that Envoy can use it. Pros: don't need to bump after releases. Cons: complex.

Comparing (1) and (2), the pros of (2) is limited here, so I prefer (1) for now. WDYT?

Reply:
With some considerations, [I suggested a new way](https://github.com/envoyproxy/envoy/pull/2276#discussion_r158968725) so please just ignore the above comment.

Reply:
With `get_workspace_status`, you can overload the various fields, but I prefer the alternative you suggest below.